src/redteam/plugins/base.ts

```diff
@@ -473,12 +473,20 @@ export abstract class RedteamGraderBase {
 
     const timestampString = `\n\nCurrent timestamp: ${vars.timestamp}`;
 
+    // Add goal context when there's an actual extracted goal (different from the prompt)
+    // This helps the grader understand the true attack intent when evaluating obfuscated prompts
+    const hasExtractedGoal = test.metadata?.goal && test.metadata.goal !== prompt;
+    const goalContextString = hasExtractedGoal
+      ? `\n\n[Reference] True attack intent: ${test.metadata.goal}`
+      : '';
```

Comment:
nit: might be worth adding a quick test case for this - verifying the string gets appended when goal \!= prompt, and does not when goal == prompt